x-pack/plugins/ecs_data_quality_dashboard/server/routes/results/post_results.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IRouter, Logger } from '@kbn/core/server';
+
+import { RESULTS_ROUTE_PATH, INTERNAL_API_VERSION } from '../../../common/constants';
+import { buildResponse } from '../../lib/build_response';
+import { buildRouteValidation } from '../../schemas/common';
+import { PostResultBody } from '../../schemas/result';
+import { API_DEFAULT_ERROR_MESSAGE } from '../../translations';
+import type { DataQualityDashboardRequestHandlerContext } from '../../types';
+import { createDocumentFromResult } from './parser';
+import { API_RESULTS_INDEX_NOT_AVAILABLE } from './translations';
+
+export const postResultsRoute = (
+  router: IRouter<DataQualityDashboardRequestHandlerContext>,
+  logger: Logger
+) => {
+  router.versioned
+    .post({
+      path: RESULTS_ROUTE_PATH,
+      access: 'internal',
+    })
+    .addVersion(
+      {
+        version: INTERNAL_API_VERSION,
+        validate: {
+          request: {
+            body: buildRouteValidation(PostResultBody),
+          },
+        },
+      },
+      async (context, request, response) => {
+        const { dataQualityDashboard } = await context.resolve(['core', 'dataQualityDashboard']);
+        const { getResultsIndexName } = dataQualityDashboard;
+        const resp = buildResponse(response);
+
+        let index: string;
+        try {
+          index = await getResultsIndexName();
+        } catch (err) {
+          logger.error(JSON.stringify(err));
```

Comment:
Consider logging additional context (for example, the route name) here, and in the other `catch` block in the this file below, because it appears this [may log an empty object in some cases](https://stackoverflow.com/questions/18391212/is-it-not-possible-to-stringify-an-error-using-json-stringify), per the following simulated example:

```ts
        let index: string;
        try {
          throw new Error('simulated error');
          index = await getResultsIndexName();
        } catch (err) {
          console.log('--> before error');
          logger.error(JSON.stringify(err));
          console.log('--> after error');
          return resp.error({
            body: `${API_RESULTS_INDEX_NOT_AVAILABLE}: ${err.message}`,
            statusCode: 503,
          });
        }
```

the code above logs the following (server side) output:

```
--> before error
[2024-01-16T17:37:04.051-05:00][ERROR][plugins.ecsDataQualityDashboard] {}
--> after error
```

In the example log output above, an empty object (with no additional context other than the plugin name) is logged.

Reply:
Context included to the error text, result:

```
[2024-01-17T11:24:16.960+01:00][ERROR][plugins.ecsDataQualityDashboard] [POST result] Error retrieving results index name: simulated error
```

Added tests as well.